storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -609,14 +637,46 @@ private SafeFuture<Optional<StateAndBlockSummary>> getOrRegenerateBlockAndState(
     if (cachedResult.isPresent()) {
       return SafeFuture.completedFuture(cachedResult);
     }
+
+    // is it an epoch boundary?
+    if (maybeEpochStates.isPresent()) {
```

Comment:
what about?

```java
    final Optional<StateAndBlockSummary> maybeEpochState =
        maybeEpochStates.flatMap(epochStates -> Optional.ofNullable(epochStates.get(blockRoot)));
    if (maybeEpochState.isPresent()) {
      return SafeFuture.completedFuture(maybeEpochState);
    }
```

Reply:
i really dont mind, can change to that format i guess.